src/Linters/UnusedVariableLinter.hack

```diff
@@ -21,6 +21,11 @@ final class UnusedVariableLinter extends AutoFixingASTLinter {
     VariableExpression $node,
   ): ?ASTLintError {
 
+    $parent = $functionish->getParentOfDescendant($node);
+    if ($parent is MemberSelectionExpression) {
```

Comment:
This is probably the best way for now. At some point it would probably be worth adding a quick "don't recurse into this subtree" interface to `ASTLinter` just for performance, but this is fine.

Could you copy the explanation from your PR into inline code comments here please?